src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2321,7 +2321,7 @@ bool IsMethodNeededByTypeDueToPreservedScope (MethodDefinition method)
 				// If the type is marked, we need to keep overrides of abstract members defined in assemblies
 				// that are copied.  However, if the base method is virtual, then we don't need to keep the override
 				// until the type could be instantiated
-				if (!@base.IsAbstract)
+				if (!(@base.IsAbstract || (@base.IsStatic && @base.IsVirtual)))
```

Comment:
Not commenting on this fix because I don't have enough context anymore, but the comment 7 lines above:

```
// Just because the type is marked does not mean we need interface methods.
// if the type is never instantiated, interfaces will be removed - but only if the optimization is enabled
```

indicates that either the comment is incorrect, or the logic in the foreach loop is incorrect for static methods - instantiating a type has no effect on whether the static methods are usable. We should probably not be in the foreach.

Reply:
I agree with Michal on this one - the comment and code don't match my expectation.

Reply:
@vitek-karas and I discussed this and came to an understanding of what this method is meant to do. It has been updated with a matrix of when to mark a method or leave it unmarked. This method also assumes that the type is not instantiated.